File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/ruby/handlers.rb
require_relative 'service'
require_relative 'models'

def handle_create_dog(request)
  name = request["name"]
  create_dog(name)
end

def handle_create_cat(request)
  name = request["name"]
  create_cat(name)
end

def handle_transfer(request)
  shelter = Shelter.new
  dog = Dog.new(request["name"])
  transfer_animal(dog, shelter)
  shelter.count
end

def handle_list(request)
  list_animals
end

def validate(request)
  raise ArgumentError, "name required" unless request["name"]
  true
end

File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/cpp/handlers.cpp
#include "models.hpp"
#include "service.cpp"

Dog handleCreateDog(Request* request) {
    std::string name = request->get("name");
    return createDog(name);
}

Cat handleCreateCat(Request* request) {
    std::string name = request->get("name");
    return createCat(name);
}

int handleTransfer(Request* request) {
    Shelter shelter;
    Dog dog(request->get("name"));
    transferAnimal(&dog, &shelter);
    return shelter.count();
}

void* handleList(Request* request) {
    return listAnimals();
}

bool validate(Request* request) {
    if (request->get("name").empty()) {
        throw std::invalid_argument("name required");
    }
    return true;
}
